### ISSUE TITLE
CatBoost is much slower than GBM LightGBM or XGBoost?

### ISSUE THREAD

**Issue by jmq19950824** [catboost/catboost#795]
Problem: {Here I use cv methods to train the following models and I found the CatBoost is much slower than the alternative methods, including GBM LightGBM and XGBoost
My training set has 1200 rows and 51 features. and I use almost the same grid to search and select the best estimators.

        #define the grid search method
        def GridSearch(clf,params,X,y):
            #here we use two cores and 10-folds
            model = GridSearchCV(clf, params, scoring='accuracy',n_jobs=-1,cv=5)
            model.fit(X, y)
            
            print(model.best_params_)
            #return the best estimator
            return model.best_estimator_

        #2.2 GBM
        params_grid_GBM=[{'learning_rate':[0.05,0.1,0.15],
                          'max_depth':[1,3,5],
                          'n_estimators':[50,100,150]}]
        Model_Ensemble_GBM=GridSearch(GradientBoostingClassifier(random_state=seed),params_grid_GBM,X_train,y_train)
        

        #2.3 LightGBM
        params_grid_LightGBM=[{'colsample_bytree':[0.6,0.8],
                               'learning_date':[0.05,0.1,0.15],
                               'max_depth':[1,3,5],
                               'n_estimators':[50,100,150]}]
        Model_Ensemble_LightGBM=GridSearch(lgb.LGBMClassifier(random_state=seed),params_grid_LightGBM,X_train,y_train)
   
        #2.4 XGBoost
        params_grid_XGBoost=[{'colsample_bytree':[0.6,0.8],
                              'learning_rate':[0.05,0.1,0.15],
                              'max_depth':[1,3,5],
                              'n_estimators':[50,100,150]}]
        Model_Ensemble_XGBoost=GridSearch(xgb.XGBClassifier(random_state=seed),params_grid_XGBoost,X_train,y_train)
     
        #2.5 CatBoost
        params_grid_CatBoost=[{'rsm':[0.6,0.8],
                               'learning_rate':[0.05,0.1,0.15],
                               'max_depth':[1,3,5],
                               'n_estimators':[50,100,150]}]
        Model_Ensemble_CatBoost=GridSearchCV(cb.CatBoostClassifier(random_state=seed,thread_count=-1,verbose=False),param_grid=params_grid_CatBoost,
                                             scoring='accuracy',cv=5).fit(X_train,y_train)
}
catboost version: {version}
Operating System: {Windows 10}
CPU: {CPU}

**Comment by annaveronika** [comment]
CatBoost has several things that LightGBM and XGBoost do not have, that you can disable.
1. First, set boosting_type='Plain' - this will be classical gradient boosting instead of 'Ordered' one. For large datasets Plain is the default mode, but for small ones Ordered might give better results.

2. Second, set leaf_estimation_iterations=1 - this will perform a single newton step inside one tree.
This is a bottleneck if you have not a lot of features. If there are many features, this will be the same.

3. We do some preprocessing of the data, which is a bottleneck if you have little amount of iterations, like 50 iterations. It is never a good idea to run boosting for 50 iterations - you need at least few hundreds to convergence.

We care about quality first, so we don't disable these things in catboost by default - they give quality boost. And we also don't tune speed for little amount of iterations, because it is necessary to train a larger ensabmle to get a decent quality.

**Comment by annaveronika** [comment]
@jmq19950824 Please, let us know if this helps!

**Comment by jmq19950824** [comment]
> @jmq19950824 Please, let us know if this helps!

Thanks a lot! I will figure that out! Anyway, thank for your suggestion and I will try the hyper-parameters aforemetioned. Please close this issue.

**Comment by miaojunlee** [comment]
Does catboost have the n_jobs=-1 like what lightgbm does?

**Comment by annaveronika** [comment]
it has thread_count=-1 which is the same
